File: core/tests/fixtures/mothers/borrow_line_mother.py
from src.domain.borrow.borrow_line import BorrowLine
from tests.fixtures.mothers.base import BaseMother
from .borrow_mother import BorrowMother
from .book_mother import BookMother


class BorrowLineMother(BaseMother):
    """Mother class for BorrowLine entity."""

    _ROMEO_AND_JULIET = {
        'borrow': BorrowMother.john_doe,
        'book': BookMother.romeo_and_juliet,
    }

    _DON_QUIJOTE = {
        'borrow': BorrowMother.john_doe,
        'book': BookMother.don_quijote,
    }

    @classmethod
    def romeo_and_juliet(cls, **overrides) -> BorrowLine:
        """Returns a BorrowLine instance for Romeo and Juliet with optional overrides."""
        return cls._create(cls._ROMEO_AND_JULIET, overrides)

    @classmethod
    def don_quijote(cls, **overrides) -> BorrowLine:
        """Returns a BorrowLine instance for Don Quixote with optional overrides."""
        return cls._create(cls._DON_QUIJOTE, overrides)

    @classmethod
    def _create(cls, values: dict, overrides: dict) -> BorrowLine:
        """Creates a BorrowLine instance with given values and overrides."""
        data = cls._merge(values, overrides)
        return BorrowLine.create(**data)

File: editor/tests/fixtures/mothers/borrow-line.mother.ts
import { BookDescriptorMother } from '@tests/fixtures/mothers/book.mother';
import { BaseMother } from '@tests/fixtures/mothers/base/base-mother';
import { BorrowLine } from '@/borrows/model/borrow-line';

const ROMEO_AND_JULIET = {
  lineId: 'cca6ad12-379d-4107-b460-dd1cb6506cb0',
  book: BookDescriptorMother.romeoAndJuliet,
  returned: false,
  returnedDate: null,
  penalty: false,
  penaltyAmount: 0,
};

const DON_QUIXOTE = {
  lineId: '752fafed-f1f2-4868-a452-a1db7acdfdbd',
  book: BookDescriptorMother.donQuixote,
  returned: false,
  returnedDate: null,
  penalty: false,
  penaltyAmount: 0,
};

type BorrowLineFixture = typeof ROMEO_AND_JULIET;

export class BorrowLineMother extends BaseMother {
  static romeoAndJuliet(
    this: void,
    overrides?: Partial<BorrowLine>,
  ): BorrowLine {
    return BorrowLineMother.create(ROMEO_AND_JULIET, overrides);
  }

  static donQuixote(this: void, overrides?: Partial<BorrowLine>): BorrowLine {
    return BorrowLineMother.create(DON_QUIXOTE, overrides);
  }

  protected static create(
    values: BorrowLineFixture,
    overrides?: Partial<BorrowLine>,
  ) {
    const fields: BorrowLine = BorrowLineMother.merge<
      BorrowLineMother,
      BorrowLine
    >(values, overrides);

    return BorrowLine.from(fields);
  }
}
